SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -3182,12 +3263,94 @@ FUnrealObjectRef USpatialNetDriver::GetCurrentPlayerControllerRef()
 
 void USpatialNetDriver::PushCrossServerRPCSender(AActor* SenderActor)
 {
-	GSenderStack.Add(SenderActor);
+	using namespace SpatialNetDriverPrivate;
+	check(!GSenderActor.IsSet());
+	GSenderActor.Emplace(SenderActorDesc(SenderActor, SenderActorDesc::Sender));
+}
+
+void USpatialNetDriver::PopCrossServerRPCSender()
+{
+	using namespace SpatialNetDriverPrivate;
+	GSenderActor.Reset();
+}
+
+void USpatialNetDriver::PushDependentActor(AActor* Dependent)
```

Comment:
Why distinguish between dependent and sender? Can we unify the terms somehow if sender isn't suitable always. Instigator?

Reply:
Instigator is already used for another purpose.
The distinction is there to allow us to do the sanity check around the API usage, since most of the diagnostic code is in the NetDriver.